packages/storage-plus/src/indexed_map.rs

```diff
@@ -793,6 +826,35 @@ mod test {
         assert_eq!(datas[1], marias[1].1);
     }
 
+    #[test]
+    fn unique_index_composite_key_range_de() {
+        let mut store = MockStorage::new();
+        let map = build_map();
+
+        // save data
+        let (pks, datas) = save_data(&mut store, &map);
+
+        let res: StdResult<Vec<_>> = map
+            .idx
+            .name_lastname
+            .prefix(b"Maria".to_vec())
+            .range_de(&store, None, None, Order::Ascending)
```

Comment:
`Prefix` must work with the key of the index in this case (that's what you are prefixing). In this impl, what must be changed is the return value of `Prefix`. Instead of being `Suffix`, it has to be `PK` (for `UniqueIndex`).

Not sure about this, by the way. Will have to take a detailed look to answer further, in a more detailed or precise way.

What we can do is to leave this on hold, documenting the issues, and take a second attempt at it in the future.